segmentation/utils/signal_extraction.py

```diff
@@ -1,31 +1,95 @@
 import numpy as np
 
 
-def simple_weighted_extraction(cell_coords, image_data):
+# TODO: work on weighted extraction and implement other discussed techniques of extraction
+def positive_pixels_extraction(cell_coords, image_data):
     """
-    Define a weighted extraction technique based on how close the coordinate is to the center
+    Extract channel counts by summing over the number of non-zero pixels in the cell,
+    improves on default_extraction by not distinguishing between pixel expression values
 
     Args:
-        cell_coords: tuples of values representing pixels within one cell
-        image_data: array containing channel counts
+        cell_coords (numpy): values representing pixels within one cell
+        image_data (xarray): array containing channel counts
 
     Returns:
-        np.array: sum of counts for each channel
+        channel_counts (numpy): sum of counts for each channel
     """
 
-    pass
+    # transpose coords so they can be used to index an array
+    cell_coords = cell_coords.T
+    channel_values = image_data.values[tuple(cell_coords)]
+
+    # sum up based on a binary mask that is 1 if the expression value > 0 else 0
+    channel_counts = np.sum(channel_values > 0, axis=0)
+
+    return channel_counts
+
+
+def center_weighting_extraction(cell_coords, image_data):
+    """
+    Extract channel counts by summing over weighted expression values based on distance from center,
+    improves upon default extraction by including a level of certainty/uncertainty
+
+    Args:
+        cell_coords (numpy): values representing pixels within one cell
+        image_data (xarray): array containing channel counts
+
+    Returns:
+        channel_counts (numpy): sum of counts for each channel
+    """
+
+    # create a weighting matrix, we will assign the highest value at the center
+    # and decrease by 1 each layer we move away from it
+    # I'm going to be assume we get square matrices in image_data
+    center = int(image_data.shape[0] / 2)
+
+    # this will ensure we never get zero or negative weighting values in our weight matrix
+    center_weight = center + 1
+
+    # now build the weight matrix using ogrid
+    image_row_index, image_col_index = np.ogrid[:image_data.shape[0], :image_data.shape[1]]
+    weight_matrix = center_weight - np.maximum(np.abs(image_row_index - center), np.abs(image_col_index - center))
+
+    # TODO: the center of even-length arrays is not a single pixel which currently causes
+    # minor signal irregularities at different layers
```

Comment:
Once we have a segmentation mask of the pixels belonging to each cell, we can modify this to use the center of the actual cell, and weight only the pixels belonging to the cell.